psgo.go

```diff
@@ -53,35 +50,37 @@ type aixFormatDescriptor struct {
 	procFn processFunc
 }
 
-// parseDescriptors parses the input string and returns a correspodning array
-// of aixFormatDescriptors, which are expected to be separated by commas.
-// The input format is "desc1, desc2, ..., desN" where a given descriptor can be
-// specified both, in the code and in the normal form.  A concrete example is
-// "pid, %C, nice, %a".  If the input string is empty, the `DefaultFormat` is
-// used.
-func parseDescriptors(input string) ([]aixFormatDescriptor, error) {
-	if len(input) == 0 {
-		input = DefaultFormat
+// translateDescriptors parses the descriptors and returns a correspodning slice of
+// aixFormatDescriptors.  Descriptors can be specified in the normal and in the
+// code form (if supported).  If the descriptors slice is empty, the
+// `DefaultDescriptors` is used.
+func translateDescriptors(descriptors []string) ([]aixFormatDescriptor, error) {
+	if len(descriptors) == 0 {
+		descriptors = DefaultDescriptors
 	}
 
 	formatDescriptors := []aixFormatDescriptor{}
-	for _, s := range strings.Split(input, ",") {
-		s = strings.TrimSpace(s)
+	for _, d := range descriptors {
+		d = strings.TrimSpace(d)
 		found := false
-		for _, d := range descriptors {
-			if s == d.code || s == d.normal {
-				formatDescriptors = append(formatDescriptors, d)
+		for _, aix := range aixFormatDescriptors {
```

Comment:
Could be setup as a map to make this faster.

Reply:
Great idea! I'll add commit on top of the current one.